gdsfactory/components/text_rectangular.py

```diff
@@ -25,12 +26,13 @@ def text_rectangular(
         position: coordinate.
         justify: left, right or center.
         layer: for text.
+        font: function that returns dictionary of characters.
     """
     pixel_size = size
     xoffset = position[0]
     yoffset = position[1]
     component = gf.Component()
-    characters = rectangular_font()
+    characters = font()
 
     for line in text.split("\n"):
         for character in line:
```

Comment:
@joamatab The hardcoded `6` values in this loop assume a pixel width/height of 5x5. Our custom font uses 7x7 characters - can you modify this to support different character sizes as well?

Reply:
> @joamatab The hardcoded `6` values in this loop assume a pixel width/height of 5x5. Our custom font uses 7x7 characters - can you modify this to support different character sizes as well?

This is the code in 7.x that was not included in 8.x specifically verified not included in  8.0.0
https://github.com/gdsfactory/gdsfactory/pull/2471/files

Part of that code automatically detects the property of the font being used, so additional font sizes can be easily added.
https://github.com/gdsfactory/gdsfactory/pull/2471/files#diff-f3f7f9f60670ce617d7d6e9c0ab2d855778cf325a52a98ac00a9aa2f6467d567R44